forecasttools/data.py

```diff
@@ -4,108 +4,93 @@
 hospitalization counts, NHSN
 (as of 2024-09-26) COVID-19
 hospitalization counts, and
-a forecasts for a single
-jurisdiction.
+an example FluSight submission.
 """
 
 import os
+from urllib import error, request
 
-os.environ["JAX_TRACEBACK_FILTERING"] = "off"
+import polars as pl
 
-import os
 
-import arviz as az
-import jax.numpy as jnp
-import jax.random as jr
-import matplotlib.pyplot as plt
-import numpy as np
-import numpyro
-import numpyro.distributions as dist
-import patsy
-import polars as pl
-import requests
+def check_url(url: str) -> bool:
+    """
+    Checks whether a URL is valid.
+    Used for checking the links to the
+    example FluSight submission and the
+    US 2020 Census data.
+
+    url
+        The url to be checked.
+    """
+    try:
+        response = request.urlopen(url)
+        return response.status == 200
+    except error.URLError as e:
+        print(f"URL Error: {e.reason}")
+        return False
+    except error.HTTPError as e:
+        print(f"HTTP Error: {e.code}")
+        return False
+
+
+def check_file_save_path(file_save_path: str) -> None:
+    """
+    Checks whether a file path is valid.
+
+    file_save_path
+        The file path to be checked.
+    """
+    directory = os.path.dirname(file_save_path)
+    if not os.path.exists(directory):
+        raise FileNotFoundError(f"Directory does not exist: {directory}")
+    if not os.access(directory, os.W_OK):
+        raise PermissionError(f"Directory is not writable: {directory}")
+    if os.path.exists(file_save_path):
+        raise FileExistsError(f"File already exists at: {file_save_path}")
 
 
 def make_census_dataset(
-    save_directory: str,
-    file_save_name: str,
-    create_save_directory: bool = False,
+    file_save_path: str,
 ) -> None:
     """
     Retrieves US 2020 Census data in a
-    three column polars dataframe, then
+    three column Polars dataframe, then
     saves the dataset as a csv in a given
     directory, if it does not already exist.
```

Comment:
Should we have an option to overwrite the file?

Reply:
Not sure what you mean here. Is the question whether there ought to be a function for users to select what they want from the 2020 US Census data (existing csv or as part of the above functioN) OR whether the default US Census extracted data file will _ever_ really change OR whether re-running the function should, by default, overwrite any file at the save path or instead abort the operation if the file at the save path already exists?

Reply:
> re-running the function should, by default, overwrite any file at the save path or instead abort the operation if the file at the save path already exists?

I was suggesting there be an option to do this, but not do it by default. It's probably so infrequently used as to not be a problem.

Reply:
I will make an issue addressing this and further utility expansion of the `data.py` file.

Reply:
See #20